src/app/containers/PageContainer/PageContainer.tsx

```diff
@@ -0,0 +1,108 @@
+import React, {
+  useCallback,
+  useLayoutEffect,
+  useMemo,
+  useReducer,
+} from 'react';
+import merge from 'lodash.merge';
+import cloneDeep from 'lodash.clonedeep';
+import { useHistory } from 'react-router-dom';
+import { useActions } from './context/actions';
+import { PageContext, initialState } from './context/PageContext';
+import { reducer } from './context/reducer';
+import {
+  FooterTypes,
+  HeaderTypes,
+  PageContextState,
+  PageOptions,
+} from './types';
+import { usePageViews } from 'app/hooks/useAnalytics';
+import { DefaultHeaderComponent } from './components/DefaultHeaderComponent/DefaultHeaderComponent';
+import { Footer } from './components/DefaultFooterComponent/DefaultFooterComponent';
+import { FastBtcHeader } from 'app/pages/FastBtcPage/components/FastBtcHeader';
+import UserWallet from 'app/pages/BridgeDepositPage/components/UserWallet';
+
+type HeaderContainerProps = {
+  pageOptions: PageOptions;
+};
+
+export const PageContainer: React.FC<Partial<HeaderContainerProps>> = ({
+  children,
+  pageOptions,
+}) => {
+  const [state, dispatch] = useReducer(
+    reducer,
+    merge(cloneDeep(initialState), { options: pageOptions }),
+  );
+  const actions = useActions(dispatch);
+  const memoizedState = useMemo(() => state as PageContextState, [state]);
+  const { options } = memoizedState;
+
+  const history = useHistory();
+
+  const resolveOptions = useCallback(
+    (pathname: string) => {
+      const initialOptions: PageOptions = merge(
+        cloneDeep(initialState.options),
+        pageOptions,
+      );
+      if (pathname.startsWith('/fast-btc')) {
+        initialOptions.header = HeaderTypes.FAST_BTC;
+        initialOptions.footer = FooterTypes.NONE;
+      } else if (pathname.startsWith('/cross-chain')) {
+        initialOptions.header = HeaderTypes.CROSS_CHAIN;
+        initialOptions.footer = FooterTypes.NONE;
+      } else if (pathname.startsWith('/perpetuals')) {
+        initialOptions.header = HeaderTypes.LABS;
+        initialOptions.footer = FooterTypes.DEFAULT;
+      } else {
+        initialOptions.header = HeaderTypes.DEFAULT;
+        initialOptions.footer = FooterTypes.DEFAULT;
+      }
+      actions.clearOptions(initialOptions);
+    },
+    [actions, pageOptions],
+  );
+
+  // reset options to default when route changes
+  useLayoutEffect(() => {
```

Comment:
There is no need for this to run after rendering.
```suggestion
  useEffect(() => {
```